client/lib/reader-lists-tags/store.js

```diff
@@ -0,0 +1,94 @@
+// Reader Lists Tag Subscription Store
+
+// External dependencies
+import { OrderedSet, fromJS } from 'immutable';
+
+// Internal dependencies
+import { action as actionTypes } from 'lib/reader-lists-tags/constants';
+import { createReducerStore } from 'lib/store';
+
+const initialState = {
+	tags: OrderedSet(), // eslint-disable-line new-cap
```

Comment:
why one set of all tags? that forces list population to be O(n). If we had a Map< list_id:int, Tag >, lookup would be O(log32 N), and we could keep the list the same instance across gets, which would let us avoid re-renders in pure components.